flink-runtime/src/main/java/org/apache/flink/runtime/topology/SimpleTopology.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.topology;
+
+/**
+ * Base topology for all logical and execution topologies.
+ * A topology consists of {@link Vertex} and {@link Result}.
+ */
+public interface SimpleTopology<VID extends VertexID, RID extends ResultID,
```

Comment:
How about name it as `BaseTopology`?
I feel that in Flink `Simple` are usually used for a simple implementation of interfaces, such as SimpleSlotProvider, SimpleCounter.

Reply:
done